singularity-eos/eos/eos_eospac.hpp

```diff
@@ -89,86 +88,640 @@ class EOSPAC : public EosBase<EOSPAC> {
+                                        LambdaIndexer &&lambdas) const {
+    static auto const name =
+        singularity::mfuncname::member_func_name(typeid(EOSPAC).name(), __func__);
+    static auto const cname = name.c_str();
+    using namespace EospacWrapper;
+    EOS_REAL *R = const_cast<EOS_REAL *>(&rhos[0]);
+    EOS_REAL *E = const_cast<EOS_REAL *>(&sies[0]);
+    EOS_REAL *T = scratch + 0 * num;
+    EOS_REAL *dTdr = scratch + 1 * num;
+    EOS_REAL *dTde = scratch + 2 * num;
+    EOS_REAL *DEDT = dTdr;
+    EOS_REAL *DEDR = dTde;
+
+    EOS_INTEGER table = TofRE_table_;
+    {
+      EOS_INTEGER options[]{EOS_Y_CONVERT};
+      EOS_REAL values[]{sieFromSesame(1.0)};
+      EOS_INTEGER nopts = 1;
+      eosSafeInterpolate(&table, num, R, E, T, dTdr, dTde, "TofRE", Verbosity::Quiet,
+                         options, values, nopts);
+    }
+
+    table = EofRT_table_;
+    {
+      EOS_INTEGER options[]{EOS_XY_PASSTHRU};
+      EOS_REAL values[]{1.0};
+      EOS_INTEGER nopts = 1;
+      eosSafeInterpolate(&table, num, R, T, E, DEDR, DEDT, "EofRT", Verbosity::Quiet,
+                         options, values, nopts);
+    }
+
+    portableFor(
+        cname, 0, num, PORTABLE_LAMBDA(const int i) {
+          cvs[i] =
+              cvFromSesame(std::max(DEDT[i], 0.0)); // Here we do something to the data!
+        });
+  }
+
+  template <typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
+  inline void BulkModulusFromDensityTemperature(ConstRealIndexer &&rhos,
+                                                ConstRealIndexer &&temperatures,
+                                                RealIndexer &&bmods, Real *scratch,
+                                                const int num,
+                                                LambdaIndexer &&lambdas) const {
+    static auto const name =
+        singularity::mfuncname::member_func_name(typeid(EOSPAC).name(), __func__);
+    static auto const cname = name.c_str();
+    using namespace EospacWrapper;
+    EOS_REAL *R = const_cast<EOS_REAL *>(&rhos[0]);
+    EOS_REAL *T = const_cast<EOS_REAL *>(&temperatures[0]);
+    EOS_REAL *E = scratch + 0 * num;
+    EOS_REAL *P = scratch + 1 * num;
+    EOS_REAL *DPDT = scratch + 2 * num;
+    EOS_REAL *DPDR = scratch + 3 * num;
+    EOS_REAL *DEDT = scratch + 4 * num;
+    EOS_REAL *DEDR = scratch + 5 * num;
+
+    EOS_INTEGER table = EofRT_table_;
+    {
+      EOS_INTEGER options[]{EOS_XY_PASSTHRU};
+      EOS_REAL values[]{1.0};
+      EOS_INTEGER nopts = 1;
+      eosSafeInterpolate(&table, num, R, T, E, DEDR, DEDT, "EofRT", Verbosity::Quiet,
+                         options, values, nopts);
+    }
+
+    table = PofRT_table_;
+    {
+      EOS_INTEGER options[]{EOS_XY_PASSTHRU};
+      EOS_REAL values[]{1.0};
+      EOS_INTEGER nopts = 1;
+      eosSafeInterpolate(&table, num, R, T, P, DPDR, DPDT, "PofRT", Verbosity::Quiet,
+                         options, values, nopts);
+    }
```

Comment:
Just a comment for the future for all these derivative quantities that require two lookups: we'll want to work with the EOSPAC team to find more performant ways to do this. If you _just_ want the bulk modulus and nothing else, we can probably accomplish this with a single lookup and let EOSPAC handle the thermodynamics to figure out how to do it.

Again... no changes needed as this is a future consideration and I'll make an issue.